kong/plugins/datadog/log.lua

```diff
@@ -0,0 +1,71 @@
+local statsd_logger = require "kong.plugins.datadog.statsd_logger"
+
+local ngx_log = ngx.log
+local ngx_timer_at = ngx.timer.at
+local string_gsub = string.gsub
+
+local _M = {}
+
+local METRICS = { "request_count", "latency", "request_size", "status_count" }
+
+local function request_counter(api_name, logger)
+  local stat = api_name..".request.count"
+  logger:counter(stat, 1, 1)
+end
+
+local function status_counter(api_name, message, logger)
+  local stat = api_name..".request.status."..message.response.status
+  logger:counter(stat, 1, 1)
+end
+
+local function request_size_gauge(api_name, message, logger)
+  local stat = api_name..".request.size"
+  logger:gauge(stat, message.request.size, 1)
+end
+
+local function latency_gauge(api_name, message, logger)
+  local latency = message.response.response_received - message.started_at
```

Comment:
To compute this, you can avoid doing what you do in `handler.lua`, and do:

```
(ngx.ctx.KONG_PROXY_LATENCY or 0) + -- time taken by Kong to chose where to proxy
(ngx.ctx.KONG_WATING_TIME or 0) + -- time elapsed while waiting for the response from upstream
(ngx.ctx.KONG_RECEIVE_TIME or 0) -- time spent on receiving the response
```

But I must ask, is this really what you want here? By "latency", I would rather expected a value that indicates how much overload Kong added to the raw API. (Which would be just `ngx.ctx.KONG_PROXY_LATENCY`). Otherwise this is more like the total time taken for a request to be proxied and sent back to the client. So is this the desired value?